src/intrinsics/MapPrototype.mts

```diff
@@ -118,6 +119,73 @@ function MapProto_get([key = Value.undefined]: Arguments, { thisValue }: Functio
   return Value.undefined;
 }
 
+/**  https://tc39.es/proposal-upsert/#sec-map.prototype.getOrInsert */
+function MapProto_getOrInsert([key = Value.undefined, value = Value.undefined]: Arguments, { thisValue }: FunctionCallContext): ValueCompletion {
+  // 1. Let M be the this value.
+  const M = thisValue as MapObject;
+  // 2. Perform ? RequireInternalSlot(M, [[MapData]]).
+  Q(RequireInternalSlot(M, 'MapData'));
+  // 3. Set key to CanonicalizeKeyedCollectionKey(key).
+  key = CanonicalizeKeyedCollectionKey(key);
+  // 4. For each Record { [[Key]], [[Value]] } p of M.[[MapData]], do
+  const entries = M.MapData;
+  for (const p of entries) {
+    // a. If p.[[Key]] is not empty and SameValue(p.[[Key]], key) is true, return p.[[Value]].
+    if (p.Key !== undefined && SameValue(p.Key, key) === Value.true) {
+      return p.Value!;
+    }
+  }
+  // 5. Let p be the Record { [[Key]]: key, [[Value]]: value }.
+  const p = { Key: key, Value: value };
+  // 6. Append p to M.[[MapData]].
+  entries.push(p);
+  // 7. Return value.
+  return value;
+}
+
+/**  https://tc39.es/proposal-upsert/#sec-map.prototype.getOrInsertComputed */
+function* MapProto_getOrInsertComputed([key = Value.undefined, callbackfn = Value.undefined]: Arguments, { thisValue }: FunctionCallContext): ValueEvaluator {
+  // 1. Let M be the this value.
+  const M = thisValue as MapObject;
+  // 2. Perform ? RequireInternalSlot(M, [[MapData]]).
+  Q(RequireInternalSlot(M, 'MapData'));
+  // 3. If IsCallable(callbackfn) is false, throw a TypeError exception.
+  if (!IsCallable(callbackfn)) {
+    return surroundingAgent.Throw('TypeError', 'NotAFunction', callbackfn);
+  }
+  // 4. Set key to CanonicalizeKeyedCollectionKey(key).
+  if (key instanceof NumberValue && Object.is(R(key), -0)) {
+    key = F(+0);
+  }
```

Comment:
and you should call `CanonicalizeKeyedCollectionKey` you just added 👀

Reply:
Oops, thanks for catching that